deep_qa/layers/entailment_models/decomposable_attention.py

```diff
@@ -6,29 +6,41 @@
     '''
-    This layer is a reimplementation of the entailment algorithm described in the following paper:
-    "A Decomposable Attention Model for Natural Language Inference", Parikh et al., 2016.
-    The algorithm has three main steps:
-
-    1) Attend: Compute dot products between all pairs of projections of words in the
-      hypothesis and the premise, normalize those dot products to use them to align each word
-      in premise to a phrase in the hypothesis and vice-versa. These alignments are then used to
-      summarize the aligned phrase in the other sentence as a weighted sum. The initial word
-      projections are computed using a feed forward NN, F.
-    2) Compare: Pass a concatenation of each word in the premise and the summary of its aligned
-      phrase in the hypothesis through a feed forward NN, G, to get a projected comparison. Do the
-      same with the hypothesis and the aligned phrase from the premise.
-    3) Aggregate: Sum over the comparisons to get a single vector each for premise-hypothesis
-      comparison, and hypothesis-premise comparison. Pass them through a third feed forward NN (H),
-      to get the entailment decision.
-
-    This layer can take either a tuple (premise, hypothesis) or a concatenation of them as input.
+    This layer is a reimplementation of the entailment algorithm described in
+    "A Decomposable Attention Model for Natural Language Inference", Parikh et
+    al., 2016. The algorithm has three main steps:
+
+    (1) Attend: Compute dot products between all pairs of projections of words
+        in the hypothesis and the premise, normalize those dot products to use
+        them to align each word in premise to a phrase in the hypothesis and
+        vice-versa. These alignments are then used to summarize the aligned
+        phrase in the other sentence as a weighted sum. The initial word
+        projections are computed using a feed forward NN, F.
+
+    (2) Compare: Pass a concatenation of each word in the premise and the
+        summary of its aligned phrase in the hypothesis through a feed forward
+        NN, G, to get a projected comparison. Do the same with the hypothesis
+        and the aligned phrase from the premise.
+
+    (3) Aggregate: Sum over the comparisons to get a single vector each for
+        premise-hypothesis comparison, and hypothesis-premise comparison. Pass
+        them through a third feed forward NN (H), to get the entailment
+        decision.
+
+    This layer can take either a tuple (premise, hypothesis) or a concatenation
+    of them as input.
+
     Expected shapes:
-        tuple input: (batch_size, sentence_length, embed_dim), (batch_size, sentence_length, embed_dim)
-        single input: (batch_size, sentence_length*2, embed_dim)
-    NOTE: premise_length = hypothesis_length = sentence_length below.
+
+    - tuple input: ``(batch_size, sentence_length, embed_dim)``, ``(batch_size, sentence_length, embed_dim)``
```

Comment:
Maybe make this match the `Inputs` format we have elsewhere?

Also, can you leave a TODO or open an issue that these `WordAlignmentEntailment` layers need to be updated?  They don't currently work correctly with loading models.